### ISSUE TITLE
LNM MathPlugin: update target if parameters changed

### ISSUE THREAD

**Issue by mhier** [ChimeraTK/DeviceAccess#186]
If the MathPlugin is used for writing, the write operation is currently only triggered when the main variable is written. Hence, if one of the parameters are updated, the target value is no longer consistent with the changed parameter.

This can be improved by checking whether the parameter(s) support wait_for_new_data. If any parameter does, the following actions should take place:

- ~~on the first write operation on the register (by _any_ accessor for the register)~~ **upon opening the device**, a new thread is launched
- the thread waits via a ReadAnyGroup until any of the parameters is changed
- when this happens, all poll-typed parameters are update, formula is evaluated based on the "x" value from the latest write operation, and the result is written to the target accessor
- write operations on the register are still executed normally (after updating all parameters as poll inputs) in the decorator
- the write operations in the decorator (including the parameter updates etc.) need to be protected with a lock against concurrent operation of the thread (also including poll parameter updates etc.)
- when the tread is launched, the launching ~~write operation~~ **open()** must block until the thread has created the ReadAnyGroup (otherwise parameter updates might be missed by the thread)
- the thread is owned by the ~~backend~~ plugin
- the thread needs to be terminated when the backend is closed or destroyed, or when the backend goes into an exception state (runtime_error resp. setException)
- when the backend is re-opened to recover, ~~the next write operation on the register will again launch the thread~~ **the thread ist started again**
- any runtime_error or logic_error seen by the thread calls setException and terminates the thread

**Comment by mhier** [comment]
updated ticket description: register writes should not change the behaviour of the backend, hence it is more sensible to enable the thread when the device is opened.

**Comment by mhier** [comment]
Implementation is somewhat complete, but it does not work properly. I have disabled the key parts of the implementation for now (via #if 0) to avoid disturbing the normal operation.

Known (but not understood) problems:
- In tests with a real application (llrfctrl server), it was observed that the _lastWrittenValue as seen by the updater thread contains huge values or even Infinity, if a non-zero value was written to the register.
- The test (part of the unified backend test) is not sensitive to this problem, in fact it is not even sensitive to the fact that the entire feature is disabled. The test register "RegVariableAsPushParameterInMath" has been removed from the test anyway.
